filebeat/prospector/prospector_log.go

```diff
@@ -54,6 +56,8 @@ func (p *ProspectorLog) Init() {
 }
 
 func (p *ProspectorLog) Run() {
+	p.mutex.Lock()
```

Comment:
This indicates that you suspect that Run is called for the same prospector multiple times in parallel. This should not be the case as Run is not executed inside a go routine: https://github.com/elastic/beats/blob/master/filebeat/prospector/prospector.go#L145 Let me know if I miss some parts here.